### PR TITLE
allow axiom tcs

### LINKED ISSUE #1054: Allow constants to be TCs
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Checked that your issue isn't already [filed](https://github.com/leanprover/lean4/issues).
    * Reduced the issue to a self-contained, reproducible test case.

### Description

Constants are not alowed to be type classes, which, for drafting/prototyping would be really useful.

### PR DESCRIPTION
**Pull request by javra**
Attempts to close #1054

### CODE REVIEW

**Comment by leodemoura** [comment]
@javra Thanks for the PR. Could you please fix the commit messages?

File: src/Lean/Class.lean [hunk 1]
```diff
@@ -84,7 +84,10 @@ def addClass (env : Environment) (clsName : Name) : Except String Environment :=
     | some decl@(ConstantInfo.inductInfo _) => do
       let b ← checkOutParam 1 #[] decl.type
       Except.ok (classExtension.addEntry env { name := clsName, hasOutParam := b })
-    | some _    => Except.error ("invalid 'class', declaration '" ++ toString clsName ++ "' must be inductive datatype or structure")
+    | some decl@(ConstantInfo.axiomInfo _) => do
+      let b ← checkOutParam 1 #[] decl.type
+      Except.ok (classExtension.addEntry env { name := clsName, hasOutParam := b })
```
Comment (javra) [inline src/Lean/Class.lean:89]: What's the sleekest way to avoid the code duplication here?

Comment (Kha) [reply src/Lean/Class.lean:89]: How about
```lean
def addClass (env : Environment) (clsName : Name) : Except String Environment := do
  if isClass env clsName then
    throw s!"class has already been declared '{clsName}'"
  let some decl := env.find? clsName
    | throw s!"unknown declaration '{clsName}'"
  unless decl matches .inductInfo .. || decl matches .axiomInfo .. do
    throw s!"invalid 'class', declaration '{clsName}' must be inductive datatype, structure, or constant"
  let b ← checkOutParam 1 #[] decl.type
  return classExtension.addEntry env { name := clsName, hasOutParam := b }
```

**Comment by javra** [comment]
> Thanks for the PR. Could you please fix the commit messages?

Will squash them into one. I first did not intend this to become a PR ;)

**Pull request merged**